### PR TITLE
Bug 1729030 - Swift: Implement URL metric type

### PR DESCRIPTION
**Pull request by badboy**
@Dexterp37 for visibility. @travis79 for review.

### CODE REVIEW

**Comment by badboy** [comment]
Making sure this time I added the files, from the CI run:

```
UrlMetricTypeTests
    ✓ testSettingLongURLsRecordsAnError (0.427 seconds)
    ✓ testSettingURLType (0.332 seconds)
    ✓ testUrlGetValueThrowsExceptionIfNothingIsStored (0.362 seconds)
    ✓ testUrlMustNotRecordIfDisabled (0.369 seconds)
    ✓ testUrlSavesToSecondaryPings (0.345 seconds)
    ✓ testUrlSavesToStorage (0.383 seconds)
```

**Review by Dexterp37 — APPROVED** [review]

File: glean-core/ios/GleanTests/Metrics/UrlMetricTests.swift [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+@testable import Glean
+import XCTest
+
+// swiftlint:disable force_cast
+// REASON: Used in a test
+class UrlMetricTypeTests: XCTestCase {
+    override func setUp() {
+        Glean.shared.resetGlean(clearStores: true)
```
Comment (travis79) [inline glean-core/ios/GleanTests/Metrics/UrlMetricTests.swift:12]: This has the potential to leak pings, so one of us needs to remember to fix it depending on who's PR lands first ;)

**Review by travis79 — APPROVED** [review]
LGTM!

File: glean-core/ios/GleanTests/Metrics/UrlMetricTests.swift (hunk 1 above)
Comment (badboy) [reply glean-core/ios/GleanTests/Metrics/UrlMetricTests.swift:12]: Landing this now, can you adopt that in your PR?

**Pull request merged**